### DISCUSSION TITLE
Possible New Addon API: Scratch Page Type

### DISCUSSION THREAD

**Discussion by GrahamSH-LLK** [ScratchAddons/ScratchAddons#538] [Ideas]
This API would return the current page type, for example, editor for the editor, project page for the project page, home for the home page, and so on. For unimportant pages like ideas, explore, download pages, and other things like that, it could return www-general or r2-general, or something generic like that. //cc @WorldLanguages

**Comment by retronbv** [comment]
Yes

**Comment by WorldLanguages** [comment]
Lol #23

**Comment by JonasPetri** [comment]
Good idea

**Comment by Dieforrobux** [comment]
what is scratch addons?

**Reply by TheColaber** [reply]
Scratch Addons is a chrome extension on the web store that lets you enable many things so scratch as more features or a different look.

**Comment by Explosion-Scratch** [comment]
How would this be useful?

**Reply by pufferfish101007** [reply]
> How would this be useful?

Rather than having multiple different userscripts for each type of page, you could have it all in one file, which makes it cleaner and more compact.

**Reply by GrahamSH-LLK** [reply]
And it means that people don’t have to resort to using the href over and over to decide whether to inject or not. You could even do `while (addon.tab === ‘mystuff) {do stuff}`

**Comment by Explosion-Scratch** [comment]
Makes sense. Thanks!
